internal/api/server/users/list_test.go

```diff
@@ -0,0 +1,70 @@
+package users
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/model"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+func TestList(test *testing.T) {
+	usersMap := db.MustGetServerUsers()
+
+	users := make([]*model.ServerUser, 0, len(usersMap))
+	for _, user := range usersMap {
+		users = append(users, user)
+	}
+
+	testCases := []struct {
+		email     string
+		permError bool
+	}{
+		// Invalid permissions.
+		{"server-user@test.com", true},
+		{"server-creator@test.com", true},
+
+		// Valid permissions.
+		{"server-admin@test.com", false},
+		{"server-owner@test.com", false},
+	}
+
+	for i, testCase := range testCases {
+		fields := map[string]any{
+			"user-email": testCase.email,
+			"user-pass":  util.Sha256HexFromString(*usersMap[testCase.email].Name),
+		}
+
+		response := core.SendTestAPIRequest(test, core.NewEndpoint(`server/users/list`), fields)
+		if !response.Success {
+			if testCase.permError {
+				expectedLocator := "-041"
+				if response.Locator != expectedLocator {
+					test.Errorf("Case %d: Incorrect error returned. Expected '%s', found '%s'.",
+						i, expectedLocator, response.Locator)
+				}
+			} else {
+				test.Errorf("Case %d: Response is not a success when it should be: '%v'.", i, response)
+			}
+
+			continue
+		}
+
+		if testCase.permError {
+			test.Errorf("Case %d: Did not get an expected permissions error.", i)
```

Comment:
Consider having an expected/actual here for more clarify.

Reply:
The statement tells us that we expected an error but did not get it, so there is an expected/actual in the language. This error message is consistent with similar tests, like [users/get_test.go](https://github.com/edulinq/autograder-server/blob/e45720fb2e3b3493e50c98ac04244ec886a72dc4/internal/api/users/get_test.go#L61).

Reply:
Ok you can resolve this then.